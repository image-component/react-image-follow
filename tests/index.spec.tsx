/* eslint-disable comma-dangle */
import React from 'react';
import { mount } from 'enzyme';
import ImageFollow from '../src';

describe('temp', () => {
  const testSrc =
    'https://github.com/image-component/gallery/blob/main/girl/1.jpg?raw=true';

  it('render', () => {
    const wrapper = mount(<ImageFollow src={testSrc} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('render children', () => {
    const wrapper = mount(
      <ImageFollow src={testSrc}>
        <h2>test</h2>
      </ImageFollow>,
    );
    expect(wrapper).toMatchSnapshot();
  });

  it('render classname', () => {
    const wrapper = mount(<ImageFollow src={testSrc} className="test33" />);
    expect(wrapper.find('.test33').length).toBeTruthy();
  });

  it('alt', () => {
    const wrapper = mount(<ImageFollow alt="alt" src={testSrc} />);
    const node = wrapper.find('.react-image-follow-img');
    expect(node.prop('alt')).toEqual('alt');
  });

  it('radius', () => {
    const wrapper = mount(<ImageFollow radius={20} src={testSrc} />);
    const node = wrapper.find('.react-image-follow-img');
    expect(node.prop('style')?.borderRadius).toEqual('20px');
  });

  it('style', () => {
    const wrapper = mount(
      <ImageFollow
        style={{
          color: 'red',
        }}
        src={testSrc}
      />,
    );
    const node = wrapper.find('.react-image-follow');
    expect(node.prop('style')?.color).toEqual('red');
  });
});
