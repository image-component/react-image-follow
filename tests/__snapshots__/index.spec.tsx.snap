// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`temp render 1`] = `
<ImageFollow>
  <a
    className="react-image-follow"
    onMouseLeave={[Function]}
    onMouseMove={[Function]}
    style={
      Object {
        "borderRadius": "8px",
      }
    }
    target="_blank"
  >
    <div
      className="react-image-follow-wrapper"
      style={
        Object {
          "borderRadius": "8px",
        }
      }
    >
      <img
        className="react-image-follow-img"
        style={
          Object {
            "borderRadius": "8px",
          }
        }
      />
    </div>
  </a>
</ImageFollow>
`;
